pkg/provisioningv2/rke2/planner/planner.go

```diff
@@ -801,7 +802,9 @@ func addTaints(config map[string]interface{}, machine *capi.Machine) error {
 	}
 
 	for _, taint := range taints {
-		taintString = append(taintString, fmt.Sprintf("%s=%s:%s", taint.Key, taint.Value, taint.Effect))
+		if taint.Key != "" && taint.Value != "" && taint.Effect != "" {
```

Comment:
Do we want to return an error if the taint isn't in the valid form?

Reply:
i just dont want "=:" ending up in the list of taints config